phase1/gce/do

```diff
@@ -53,9 +61,17 @@ fetch_kubeconfig() {
     # succeeds, then stdout is tainted by ssh initialization output, so we echo
     # a marker to indicate the start of the file, and delete all lines up to and
     # including it.
-    if gcloud compute ssh --project "$PROJECT" --zone "$ZONE" "$MASTER" --command "echo STARTFILE; sudo cat /etc/kubernetes/admin.conf" > ${TMP_DIR}/kubeconfig.raw; then
+    if gcloud compute ssh --project "$PROJECT" --zone "$ZONE" "$SSHUSER$MASTER" --command "echo STARTFILE; sudo cat /etc/kubernetes/admin.conf" > ${TMP_DIR}/kubeconfig.raw; then
       sed '/^STARTFILE$/,$!d;/^STARTFILE$/d' ${TMP_DIR}/kubeconfig.raw > ${CLUSTER_NAME}/kubeconfig.json
       echo Successfully fetched kubeconfig.
+
+      # kubeconfig.json fetched from master node has the user, cluster and context all hardcoded with
+      # 'kubernetes' keyword. So replace 'kubernetes' with the cluster_name.
+      sed -i "s/kubernetes/${CLUSTER_NAME}/g" ${CLUSTER_NAME}/kubeconfig.json
```

Comment:
This feels so dangerous. Is there any possibility that another random field could contain `kubernetes` and not actually refer to the context you're trying to update? I'm wondering if it's possible to make these transformations using [kubectl config](https://kubernetes-v1-4.github.io/docs/user-guide/kubectl/kubectl_config/) subcommands instead of manual sed regexes.

If we do want to do this, I would still gate this logic behind the below `if [[ -n "${KUBE_CONTEXT_NAME}" ]]` so that users would need to willingly opt into the potentially dangerous operation.

I thought when we originally discussed during the SIG Cluster Lifecycle meeting, it was a good enough solution to maintain different kubeconfig files for the different clusters. What pushed you back into the direction of wanting to merge them all as different contexts into the same kubeconfig file?